evalit/_base.py

```diff
@@ -26,10 +26,44 @@ class AbstractAutomation(ABC):
         - `nifi.nifi_automation.NifiAutomation`
         - `mft.mft_automation.MFTAutomation`
 
-    Each automation component should implement `rclone_automation` method
+    Each automation component should implement `rclone_automation(...)` method
     and must return `Tuple[TransferDTO]` data structure.
+
+    Args:
+        ```config```: ```Union[str, pathlib.Path, dict]```
+            The input yaml configuration for the transfer.
+            If path, then it will load the config from yaml loader
+            Else, incoming dictionary is used as the config
+
+        ```files```: ```Optional[Union[str, Path]]```
+            List of files to benchmark.
+            If provided, only these files will be used.
+            Else, all the files from the source bucket are used
+            (Defaults to None)
+
+        ```debug```: ```bool```
+                Flag used for debugging purpose.
+                Defaults to False
+
+    Example Structure of YAML should be:
+
+        .. code-block: yaml
+
+                ---
+                source_token: "<username>"
+                source_secret: "<password>"
+                source_s3_endpoint: "http://127.0.0.1:8080"
```

Comment:
if we were to use this in live system, would s3:// link work? or need to provide just the bucket name?

Reply:
here we are trying to emulate the S3 buckets using the MINIO server. so the real s3:// link doesn't work here
@NISH1001 please feel free to add on top of this if I miss something.

Reply:
Using minio is good and all for local testing, but in real world scenario s3 tests would also make more sense.